flytekit/experimental/eager_function.py

```diff
@@ -0,0 +1,569 @@
+
+        # ensure async context is provided
+        if "async_ctx" in kwargs:
+            kwargs.pop("async_ctx")
+
+        if getattr(self.entity, "execution_mode", None) == PythonFunctionTask.ExecutionBehavior.DYNAMIC:
+            raise EagerException(
+                "Eager workflows currently do not work with dynamic workflows. "
+                "If you need to use a subworkflow, use a static @workflow or nested @eager workflow."
+            )
+
+        if self.execution_state in {
+            ExecutionState.Mode.LOCAL_TASK_EXECUTION,
+            ExecutionState.Mode.LOCAL_WORKFLOW_EXECUTION,
+        }:
+            # If running as a local workflow execution, just execute the python function
+            try:
+                if isinstance(self.entity, WorkflowBase):
+                    out = self.entity._workflow_function(**kwargs)
+                    if inspect.iscoroutine(out):
+                        # need to handle invocation of AsyncEntity tasks within the workflow
+                        out = await out
+                    return out
+                elif isinstance(self.entity, PythonTask):
+                    out = self.entity._task_function(**kwargs)
+                    if inspect.iscoroutine(out):
+                        out = await out
+                    return out
+                else:
+                    raise ValueError(f"Entity type {type(self.entity)} not supported for local execution")
+            except Exception as exc:
+                raise EagerException(
+                    f"Error executing {type(self.entity)} {self.entity.name} with {type(exc)}: {exc}"
+                ) from exc
+
+        # this is a hack to handle the case when the task.name doesn't contain the fully
+        # qualified module name
+        entity_name = (
+            f"{self.entity._instantiated_in}.{self.entity.name}"
+            if self.entity._instantiated_in not in self.entity.name
+            else self.entity.name
+        )
+
+        if isinstance(self.entity, WorkflowBase):
+            remote_entity = self.remote.fetch_workflow(name=entity_name)
+        elif isinstance(self.entity, PythonTask):
+            remote_entity = self.remote.fetch_task(name=entity_name)
+        else:
+            raise ValueError(f"Entity type {type(self.entity)} not supported for local execution")
+
+        execution = self.remote.execute(remote_entity, inputs=kwargs, type_hints=self.entity.python_interface.inputs)
+        self._execution = execution
+
+        url = self.remote.generate_console_url(execution)
+        node = AsyncNode(self, entity_name, execution, url)
+        self.async_stack.set_node(node)
+        logger.debug(url)
+
+        for _ in range(self._n_polls):
+            execution = self.remote.sync(execution)
+            if execution.closure.phase in {WorkflowExecutionPhase.FAILED}:
+                raise EagerException(f"Error executing {self.entity.name} with error: {execution.closure.error}")
```

Comment:
this line: https://github.com/flyteorg/flytekit/pull/1579/files#diff-a9c76bab852a74a971a267146887f4e3dc32b350e127300e65120cc92d22e2b3R478-R480

should be able to run the cleanup function in case any error happens in the eager workflow